drivers/gpio/gpio-kempld.c

```diff
@@ -575,13 +576,16 @@ static int kempld_gpio_irq_init(struct device *dev,
 		return ret;
 	}
 
-	ret = gpiochip_irqchip_add_nested(chip, &kempld_irqchip, 0,
-				   handle_simple_irq, IRQ_TYPE_NONE);
-	if (ret) {
-		dev_err(dev, "could not add irqchip\n");
-		return ret;
-	}
-	gpiochip_set_nested_irqchip(chip, &kempld_irqchip, gpio->irq);
+	girq = &chip->irq;
```

Comment:
I'm not sure if only these changes are enough.... the `gpiochip_irqchip_add_nested()` was retired here: https://github.com/torvalds/linux/commit/f1f37abbe6fc2b1242f78157db76e48dbf9518ee

and here is the latest docs about gpio IRQ chips:
https://github.com/torvalds/linux/blob/master/Documentation/driver-api/gpio/driver.rst#infrastructure-helpers-for-gpio-irqchips

Don't you need to define the `irq_ack, irq_mask, irq_unmask`, etc? I don't see those in the driver....

Reply:
probably, this is why I just disable the driver in the config. I looked at other examples but since we do not have any docs it is pretty hard to fix it properly, but I'll have a look. Thanks for the links!